src/neo/Wallets/Wallet.cs

```diff
@@ -288,27 +288,28 @@ public Transaction MakeTransaction(TransferOutput[] outputs, UInt160 from = null
             }
         }
 
-        public Transaction MakeTransaction(byte[] script, UInt160 sender = null, Signer[] cosigners = null, TransactionAttribute[] attributes = null)
+        public Transaction MakeTransaction(byte[] script, Signer[] signers = null, TransactionAttribute[] attributes = null)
         {
             UInt160[] accounts;
-            if (sender is null)
+            if (signers is null || signers.Length > 0)
             {
                 accounts = GetAccounts().Where(p => !p.Lock && !p.WatchOnly).Select(p => p.ScriptHash).ToArray();
             }
             else
             {
+                var sender = signers[0].Account;
                 if (!Contains(sender))
                     throw new ArgumentException($"The address {sender} was not found in the wallet");
                 accounts = new[] { sender };
             }
             using (SnapshotView snapshot = Blockchain.Singleton.GetSnapshot())
             {
                 var balances_gas = accounts.Select(p => (Account: p, Value: NativeContract.GAS.BalanceOf(snapshot, p))).Where(p => p.Value.Sign > 0).ToList();
-                return MakeTransaction(snapshot, script, cosigners ?? Array.Empty<Signer>(), attributes ?? Array.Empty<TransactionAttribute>(), balances_gas);
+                return MakeTransaction(snapshot, script, signers ?? Array.Empty<Signer>(), attributes ?? Array.Empty<TransactionAttribute>(), balances_gas);
             }
         }
 
-        private Transaction MakeTransaction(StoreView snapshot, byte[] script, Signer[] cosigners, TransactionAttribute[] attributes, List<(UInt160 Account, BigInteger Value)> balances_gas)
+        private Transaction MakeTransaction(StoreView snapshot, byte[] script, Signer[] signers, TransactionAttribute[] attributes, List<(UInt160 Account, BigInteger Value)> balances_gas)
```

Comment:
This parameter should be `cosigners`, because it doesn't include the sender. The sender is found by this method.

Reply:
Then if we want to use a different scope for sender we need to tamper the transaction after call this method?

Reply:
Let me try to modify it.

Reply:
Done. Please review it.

Reply:
Very good!